### PR TITLE
Some more bundle size improvements

### PR DESCRIPTION
**Pull request by Windvis**
This adds a bundle analyzer setup (based on the [Embroider guide](https://github.com/embroider-build/embroider/blob/main/docs/analyzing.md#analyzing-bundles)) and adds some small bundle size improvements as well.

### CODE REVIEW

File: app/helpers/iso-formatter.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { format, parseISO } from 'date-fns';
+import { lightFormat, parseISO } from 'date-fns';
```
Comment (Windvis) [inline app/helpers/iso-formatter.ts:1]: The output of `lightFormat` is the same in our use-cases, and it's a lot smaller (3KB vs 24KB minified).

File: package.json [hunk 2]
```diff
@@ -87,7 +88,7 @@
     "ember-cli-inject-live-reload": "^2.1.0",
     "ember-cli-sass": "^11.0.1",
     "ember-cli-typescript": "^5.2.1",
-    "ember-concurrency": "^3.0.0",
+    "ember-concurrency": "^3.1.0",
```
Comment (Windvis) [inline package.json:91]: This forced a dedup of ember-concurrency, so we only ship v3 now (30Kb less).

File: app/helpers/iso-formatter.ts
Comment (Windvis) [inline app/helpers/iso-formatter.ts:1]: I don't think this file is used anymore, so we can probably just remove it?

Comment (aliokan) [reply app/helpers/iso-formatter.ts:1]: @Windvis Yes! :wastebasket:

**Review by aliokan — APPROVED** [review]

File: app/helpers/iso-formatter.ts
Comment (Windvis) [reply app/helpers/iso-formatter.ts:1]: I'll do it in a follow-up PR, I believe there were some other things that were unused as well.

**Pull request merged**